control_system_stack/setyaw/src/main.py

```diff
@@ -0,0 +1,185 @@
+
+        1. Updates the Current value of yaw - Current_yaw
+        2. Calculates error and delta_error based on 4 quadrant tangent = arctan2()
+        3. Updates feedback to be sent to the client
+            - Desired_yaw
+            - Current_yaw
+            - Error
+        4. Debug messages
+        """
+        global Current_yaw
+        global base_yaw
+        global prevError
+        global Client_goal
+        global FIRST_ITERATION
+
+        Current_yaw = dataIn.yaw
+
+        if  FIRST_ITERATION :
+    		base_yaw = Current_yaw
+    		FIRST_ITERATION = False
+
+        prevError = YAW.error
+        error = (base_yaw + Client_goal.yaw - Current_yaw)* 3.14 / 180
+        YAW.error = np.arctan2(sin(error),cos(error))*180/3.14
+        YAW.delta_error = YAW.error - prevError
+
+        # Update the feedbacks to be sent
+        self._feedback.Desired_yaw = (Client_goal.yaw + base_yaw)%360
+        self._feedback.Current_yaw = Current_yaw
+        self._feedback.Error = YAW.error
+
+        # calculate the thrust from fuzzy control
+        thrust = YAW.run()
+
+        thruster6Data.data[0] = 0.0
+        thruster6Data.data[1] = 0.0
+        thruster6Data.data[2] = 0.0
+        thruster6Data.data[3] = 0.0
+        thruster6Data.data[4] =  thrust       # Left Thruster
+        thruster6Data.data[5] = -1 * thrust   # Rigt Thruster
+
+        thruster4Data.data[0] = thruster6Data.data[0]
+        thruster4Data.data[1] = thruster6Data.data[1]
+        thruster4Data.data[2] = thruster6Data.data[4]
+        thruster4Data.data[3] = thruster6Data.data[5]
+
+        # publish thrust values to thruster converter
+        pub_thrusters4.publish(thruster4Data)
+        pub_thrusters6.publish(thruster6Data)
+
+        # Debug messages
+        # rospy.logdebug("--------")
+        # rospy.logdebug("Current Yaw : %s",round(Current_yaw,3))
+        # rospy.logdebug("Desired_yaw : %s",round(self._feedback.Desired_yaw,3))
+        rospy.loginfo("Error : %s",round(YAW.error,3))
+        # rospy.logdebug("Delta_error : %s",round(YAW.delta_error ,3))
+        # rospy.logdebug("Relative_Goal : %s",round(Client_goal.yaw,3))
+        # rospy.logdebug("Thruster data L : %s",thruster6Data.data[4])
+        # rospy.logdebug("Thruster data R : %s",thruster6Data.data[5])
+
+
+
+    def execute_cb(self, goal):
+        """
+        1. Update the goal each time client sends a new goal
+        2. Wait for first imu data to be recieved, to calculate the error.
+        3. calculate the thrust from fuzzy control and send to the thruster converter.
+        4. publishes feedback back to the client
+            - Desired_yaw
+            - Current_yaw
+            - Error
+        """
+
+        global Client_goal
+        global FIRST_ITERATION
+        global Current_yaw
+
+        # Update the goal
+        Client_goal = goal
+        FIRST_ITERATION = True
+        Current_yaw = -1
+
+        rospy.Subscriber(topicHeader.ABSOLUTE_RPY, absoluteRPY, self.imuCB)
```

Comment:
@aniket11bh Move the subscriber to the `main` block of this file. Always declare all the subscribers and the publishers in the main block for easier access.